hls-plugin-api/src/Ide/Types.hs

```diff
@@ -1028,10 +1059,18 @@ configForPlugin :: Config -> PluginDescriptor c -> PluginConfig
 configForPlugin config PluginDescriptor{..}
     = Map.findWithDefault (configInitialGenericConfig pluginConfigDescriptor) pluginId (plugins config)
 
+-- | Checks that a specific plugin is enabled
+pluginGlobalEnabled :: PluginConfig -> PluginStatus
+pluginGlobalEnabled pc = if plcGlobalOn pc
+                           then PluginEnabled
+                           else PluginDisabled DisabledByConfig
+
 -- | Checks that a given plugin is both enabled and the specific feature is
 -- enabled
-pluginEnabledConfig :: (PluginConfig -> Bool) -> PluginConfig -> Bool
-pluginEnabledConfig f pluginConfig = plcGlobalOn pluginConfig && f pluginConfig
+pluginEnabledConfig :: (PluginConfig -> Bool) -> PluginConfig -> PluginStatus
```

Comment:
It feels like we're moving towards the definition of `pluginEnabled` listing the various predicates that are tried and monoidally combining them together. So it feels a bit off to be doing that partially inside here. I'd be tempted to define

```
pluginEnabledConfig f pluginConfig = if f pluginConfig then PluginEnabled else PluginDisabled DisabledByConfig
```
and then just have
```
pluginGlobalEnabled pluginConfig <> pluginEnabledConfig codeLensesOn pluginConfig
```
or whatever at the use sites.